terraform/modules/aws-dns-records/README.md

```diff
@@ -0,0 +1,34 @@
+Terraform module: DNS records
+=============================
+
+State: __experimental__
+
+This module creates a set of DNS entries on AWS. As of now it can the following type of records:
```

Comment:
DNS entries for what purpose? What should those IPs be? Why is it useful?

there are a few assumtions here that would be nice to make explicit in the readme (please reformulate the following):

* one or multiple kubernetes servers with a public IP or load balancers fronting kubernetes servers, on which wire-server's nginx-ingress listens on port 443
* basic round-robin DNS? What happens when one machine goes down?
* this is about wire-server DNS as described on docs.wire.com / find-the-right-page.
* this does not create records for restund servers

Reply:
> there are a few assumptions here that would be nice to make explicit in the readme (please reformulate the following):

I probably did't quite understand your comment. So, can you point me to the places where this module makes any of the assumptions that you mentioned (other than the default value for `subdomains`)?

Reply:
I suppose the purpose / intention / reason for existence of the module (as seen by its default subdomains list) is as a way to create DNS records pointing to a wire-server installation. I think we don't gain much on making this module very generic, I think it's most useful when geared towards the creation of the DNS records for a wire-server installation. So specifying this in the README, to me, is useful. Would you be fine adding a sentence or two about this to the README?

It's true that in theory someone could also take this module and use it for another purpose by overriding the subdomains variable. But for that, someone needs to actually look at the source code of the module to understand that. It would be nice if the purpose and usage of this terraform module was clearer by only looking at the README.

Reply:
> actually look at the source code of the module to understand that

Even though I might not given the impression, but I'm a huge fan of writing readable & self-explanatory (and hopefully in that way more maintainable) code. Therefore, I encourage anyone who wants to use that code, also to have a quickly look into it, before using it. In this particular case, it's even not that much code. Also, since this is not compiled code, and actually used where its located, it's very accessible (copy in `.terraform`).

Why this kind of thinking? The past has taught me: *Docs are not to be trusted (outdated, flawed,misleading). Look at the code if you want to know what's going on.*

That being said, It's just my personal point of view. If we, as a team, follow a different mantra, I'd be happy to adjust.
Anyway, you have a point here. I will add a note about as to why this set of default subdomains exactly.

> It's true that in theory 

For starters, I will also use it to create records for SFT machines. With this, the *theory* becomes *practice* :)